app/grandchallenge/participants/models.py

```diff
@@ -123,3 +125,81 @@ class RegistrationQuestionGroupObjectPermission(GroupObjectPermissionBase):
     content_object = models.ForeignKey(
         RegistrationQuestion, on_delete=models.CASCADE
     )
+
+
+class RegistrationQuestionAnswer(models.Model):
+    registration_request = models.ForeignKey(
+        RegistrationRequest,
+        on_delete=models.CASCADE,
+        related_name="registration_question_answers",
+    )
+    question = models.ForeignKey(
+        RegistrationQuestion,
+        on_delete=models.CASCADE,
+        related_name="answers",
+    )
+    answer = models.JSONField(default=str, blank=True)
+
+    class Meta:
+        unique_together = (("registration_request", "question"),)
+
+    @property
+    def empty_answer(self):
+        return self.answer == self._meta.get_field("answer").get_default()
+
+    def clean(self):
+        super().clean()
+
+        if self.empty_answer and self.question.required:
+            raise ValidationError(
+                f"The question {self.question.question_text!r} requires an answer"
+            )
+
+        if not self.empty_answer and self.question.schema:
+            try:
+                JSONValidator(schema=self.question.schema)(value=self.answer)
+            except ValidationError as e:
+                raise ValidationError(
+                    {"answer": f"Incorrect format: {e.__cause__}"}
+                )
+
+    def validate_constraints(self, exclude=None):
+        super().validate_constraints(exclude=exclude)
+
+        if not exclude or (
+            "registration_request" not in exclude
+            and "challenge" not in exclude
+        ):
+            # Cannot add a database-level constraint for this, so do it here:
+            if self.question.challenge != self.registration_request.challenge:
+                raise ValidationError(
+                    "Cannot answer questions for a registration with different challenges"
+                )
+
+    def save(self, *args, **kwargs):
+        adding = self._state.adding
+        super().save(*args, **kwargs)
+        if adding:
+            self.assign_permissions()
+
+    def assign_permissions(self):
+        # Admins can view an answer
+        assign_perm(
+            "view_registrationquestionanswer",
```

Comment:
Unsure if this is totally necessary as this is never in a separate view.

Reply:
Currently, the constructs are such that `'change_challenge'`, `'add_registration_question'` and `'view_registrationquestionanswer'` perfectly overlap with the `challenge.admins_group`.

Since the `reader_studies.models.Answer` follows the same pattern and `'change_challenge'` vs`'add_registration_question'` is something we want, I don't think it is overkill to cleanly add the permissions here and now.

Reply:
See the above comment about the usage of these things. They are entirely intertwined, are never used independently, so this is unnecessary.

Reply:
Removed and simplified.